src/vehicle.rs

```diff
@@ -0,0 +1,98 @@
+use std::time::{Instant, Duration};
+
+enum Action {
+    Accelerate,
+    Deccelerate,
+    StaticSpeed
+}
+
+
+trait Vehicle {
+    fn get_length() -> f32;
+    fn get_buffer_zone() -> f32;
+    fn get_position(&self) -> f32;
+    // or (f32, f32)
+    fn get_speed(&self) -> f32;
+    fn get_acceleration(&self) -> f32;
+    fn action(&mut self, action:Action);
+    fn roll_forward_by(&mut self, duration: Duration);
+}
+
+
+struct Car {
+    length: f32,
+    buffer_zone: f32,
+    position: f32,
+    speed: f32,
+    acceleration: f32,
+    
+}
+
+// impl Car {
```

Comment:
The convention in Rust is to provide a constructor called `new` that takes whatever necessary information is required to construct the object and returns a new instance. For example, I think `position` is the only truly necessary field at this time, and so the constructor might look like this:
```rust 
impl Car {
    pub fn new(position: f32) -> Car {
        Car { position, 
              length: < standard length >,
              buffer_zone <  standard buffer zone >,
              speed: 0.0f32,
              acceleration: 0.0f32
    }
}
```
The struct construction won't be available above this module as is, because the fields of the struct are hidden, so we need to provide this function if we want external users (including from within the crate) to be able to construct a `Car`